aten/src/ATen/templates/Tensor.h

```diff
@@ -280,12 +280,13 @@ class CAFFE2_API Tensor {
   /// TensorOptions.h.
   TensorOptions options() const;
 
-  void* data_ptr() const {
-    return this->unsafeGetTensorImpl()->data();
-  }
+  template <typename T = void>
+  T * data_ptr() const;
 
   template<typename T>
-  T * data() const;
+  T * data() const {
```

Comment:
shouldn't we deprecate this?  Or is that coming in a later patch?

Reply:
It's coming in the next patch: deprecating `data<T>()` and changing all call sites of `data<T>()` to `data_ptr<T>()`.